css-easing-1/Overview.bs

```diff
@@ -125,16 +125,126 @@ The syntax for specifying an [=easing function=] is as follows:
+and an [=input progress value=] |inputProgress|,
+perform the following.
+It returns an [=output progress value=].
+
+1.   If |linearEasingFunction|'s [=linear easing function/points=] [=list/is empty=],
+     then return |inputProgress|.
+
+1.   If |linearEasingFunction|'s [=linear easing function/points=] [=list/size=] is 1,
+     then return |linearEasingFunction|'s [=linear easing function/points=][0]'s [=linear easing point/output=].
+
+1.   Plot |linearEasingFunction|'s [=linear easing function/points=] points on a graph
+     using [=linear easing point/input=] as the x-axis
+     and [=linear easing point/output=] as the y-axis.
+
+1.   Interpolate the points linearly.
+
+1.   Extend the ends of the graph to infinity using the final angle of the line at either end.
+     If the line starts/ends with two points of the same position, the line should extend along the x-axis.
+
+1.   Return the y-axis value corresponding to the x-axis value of |inputProgress|.
+
+</section>
+
+<dfn function lt="linear()">linear(<<linear-stop-list>>)</dfn>
+
+<pre class="prod">
+  <dfn>&lt;linear-stop-list></dfn> = [ <<linear-stop>> ]#
+  <dfn>&lt;linear-stop></dfn> = <<number>> && <<linear-stop-length>>?
+  <dfn>&lt;linear-stop-length></dfn> = <<percentage>>{1,2}
+</pre>
+
+<section algorithm="to create a linear easing function">
+
+To <dfn>create a linear easing function</dfn>
+from a <<linear-stop-list>> |stopList|,
+perform the following.
+It returns a [=linear easing function=].
+
+1.   Let |function| be a new [=linear easing function=].
+
+1.   Let |largestInput| be negative infinity.
+
+1.   For each |stop| in |stopList|:
+
+     1.   Let |point| be a new [=linear easing point=].
+
+     1.   [=list/Append=] |point| to |function|'s [=linear easing function/points=].
+
+     1.   Set |point|'s [=linear easing point/output=] to |stop|'s <<number>>.
+
+     1.   If |stop| has a <<linear-stop-length>>, then:
+
+          1.   Set |point|'s [=linear easing point/input=] to whichever is greater:
+               |stop|'s <<linear-stop-length>>'s first <<percentage>> as a number,
+               or |largestInput|.
+
+          1.   Set |largestInput| to |point|'s [=linear easing point/input=].
+
+          1.   If |stop|'s <<linear-stop-length>> has a second <<percentage>>, then:
+
+               1.   Let |extraPoint| be a new [=linear easing point=].
+
+               1.   [=list/Append=] |extraPoint| to |function|'s [=linear easing function/points=].
+
+               1.   Set |extraPoint|'s [=linear easing point/output=] to |stop|'s <<number>>.
+
+               1.   Set |extraPoint|'s [=linear easing point/input=] to whichever is greater:
+                    |stop|'s <<linear-stop-length>>'s second <<percentage>> as a number,
+                    or |largestInput|.
+
+               1.   Set |largestInput| to |extraPoint|'s [=linear easing point/input=].
+
+     1.   Otherwise, if |stop| is the first [=list/item=] in |stopList|,
```

Comment:
Hm, admittedly this is an edge case, but if any of the entries not at either edge contain negative or higher-than-100% percentage values (`linear(0, 0.2 -10%, 0.8 110%, 1.0)`), wouldn't it break the monotonicity?

Reply:
That's true on the start side - how about the end, `linear([...], 0.8 110%, 1.0)`? Could use `|largestInput|`, though each side behaving differently seems a bit less intuitive (Debatable, since we're already in an edge case).

Reply:
I think if the last value doesn't have a position, it should be assumed to be 100%. Does that make sense?

Reply:
Actually, let me test gradients…

Reply:
https://jsbin.com/mopojul/4/edit?html,css,output

It seems like the first value is assigned 0% if it has no position. However, it seems like the last position will be `Math.max(previousPosition, 100%)`.

My intent will be to spec the same position behaviour as gradients unless there's a good reason to do something different.

Reply:
Sounds reasonable. Thanks!